### PR TITLE
VectorizedArray: Some minor doxygen markup improvements

### PR DESCRIPTION
**Pull request by tamiko**

### CODE REVIEW

**Comment by tamiko** [comment]
I would like to group the related functions of a `VectorizedArray` a little bit because I am about to introduce a bit more :-)

File: include/deal.II/base/vectorization.h [hunk 1]
```diff
@@ -627,10 +633,14 @@ vectorized_transpose_and_store(const bool                            add_into,
 }
 
 
+//@}
+
+#ifndef DOXYGEN
```
Comment (bangerth) [inline include/deal.II/base/vectorization.h]: You exclude all of the specializations of the class from the doxygen generation. Is this on purpose?

Comment (tamiko) [reply include/deal.II/base/vectorization.h]: Yes - we never generated any official documentation where these variants get picked up, see for example [1]. There is also no additional information in the documentation of the potentially 8 specializations.

[1] https://www.dealii.org/developer/doxygen/deal.II/classVectorizedArray.html

Comment (bangerth) [reply include/deal.II/base/vectorization.h]: OK. Good to merge then!

**Pull request merged**